dataprofiler/data_readers/csv_data.py

```diff
@@ -737,4 +759,4 @@ def reload(self, input_file_path=None, data=None, options=None):
             header=self.header, delimiter=self.delimiter, quotechar=self.quotechar
         )
         super(CSVData, self).reload(input_file_path, data, options)
-        self.__init__(self.input_file_path, data, options)
+        self.__init__(self.input_file_path, data, options)  # type: ignore
```

Comment:
Can we get rid of this `#type: ignore` by resolving an issue upstream?

Reply:
The issue is that mypy doesn't like using `self.__init__` because it doesn't know which constructor will end up being called (and therefore doesn't know what types are needed). However, the current code assumes that behavior so I just told mypy to ignore that line.